torchrl/data/tensor_specs.py

```diff
@@ -30,17 +32,29 @@
 import numpy as np
 import torch
 from tensordict.tensordict import TensorDict, TensorDictBase
+from tensordict.utils import _getitem_batch_size
 
 from torchrl._utils import get_binary_env_var
 
 DEVICE_TYPING = Union[torch.device, str, int]
 
 INDEX_TYPING = Union[int, torch.Tensor, np.ndarray, slice, List]
 
+SHAPE_INDEX_TYPING = Union[
+    int,
+    range,
+    List[int],
+    np.ndarray,
+    Tuple[int, None, slice, List[int], Tuple[Any], type(...)],
+    slice,
+    None,
+    torch.Tensor,
```

Comment:
Did we test indexing with a tensor?

Reply:
We have a short test here: https://github.com/pytorch/rl/pull/1081/files#diff-9b52a375cf71a2d2d441107d14b011ba181b875085cb5724a19c7f7ac9d56568R2436.